lib/timers.js

```diff
@@ -434,6 +482,15 @@ function rearm(timer) {
 
 
 const clearTimeout = exports.clearTimeout = function(timer) {
+  // Fewer checks may be possible, but these cover everything.
+  if (async_hook_fields[kDestroy] > 0 &&
+      timer &&
+      typeof timer[async_id_symbol] === 'number' &&
+      !timer._destroyed) {
+    emitDestroy(timer[async_id_symbol]);
+    timer._destroyed = true;
+  }
```

Comment:
The logic would lead me to believe so, but I don't have a test to verify that yet